lib/config/error-codes/index.js

```diff
@@ -41,7 +41,7 @@ function checkErrors(subdomain, domain) {
 
     assert(
       !errorCodes.has(error.code),
-      `Error codes file : code ${error.code} for error '${errorName}' is not unique (domain: ${domain.code}, subdomain: ${subdomain.code}).`);
+      `Error configuration file : code ${error.code} for error '${errorName}' is not unique (domain: ${domain.code}, subdomain: ${subdomain.code}).`);
```

Comment:
You can throw the custom error directly here instead of throwing a string

Reply:
As you know this file is about checking if the json files are formatted correctly
Since this errors can be triggered before those checks are finished, I prefer not to use potentially bad formatted files to throw custom errors

Reply:
I don't think these check really apply to Kuzzle native errors. 
I agree with @Aschen , would be better to throw a native custom error and check on its subcode within the catch rather than relying on the message.